app/gui/src/dashboard/components/AriaComponents/Dialog/DialogTrigger.tsx

```diff
@@ -13,20 +13,28 @@ export interface DialogTriggerRenderProps {
   readonly open: () => void
 }
 /** Props for a {@link DialogTrigger}. */
-export interface DialogTriggerProps extends Omit<aria.DialogTriggerProps, 'children'> {
+export interface DialogTriggerProps {
   /** The trigger element. */
   readonly children: [
     React.ReactElement | ((props: DialogTriggerRenderProps) => React.ReactElement),
     React.ReactElement | ((props: DialogTriggerRenderProps) => React.ReactElement),
   ]
-  readonly onOpen?: () => void
-  readonly onClose?: () => void
+  readonly onOpen?: (() => void) | undefined
+  readonly onClose?: (() => void) | undefined
+  /** Whether the overlay is open by default (controlled). */
+  readonly isOpen?: boolean | undefined
+  /** Whether the overlay is open by default (uncontrolled). */
+  readonly defaultOpen?: boolean | undefined
+  /** Handler that is called when the overlay's open state changes. */
+  readonly onOpenChange?: ((isOpen: boolean) => void) | undefined
 }
 
 /** A DialogTrigger opens a dialog when a trigger element is pressed. */
 export function DialogTrigger(props: DialogTriggerProps) {
   const { children, onOpenChange, onOpen = () => {}, onClose = () => {} } = props
 
+  // @ts-expect-error Typescript requires to explicitly add `undefined` to the props
```

Comment:
this is only true when `exactOptionalPropertyTypes` is enabled, meaning we can disable it. i personally like it, but how valuable it is, is certainly questionable. that said, i think the only library that's causing problems is `react-aria-components`, and realistically it should ideally be fixed on *their* side. so whether we keep or disable the option, i don't know

Reply:
I think keeping this as unsafe with comment is fine, even though it's a bit annoying.